### PR TITLE
Feature details fragment

### PR DESCRIPTION
**Pull request by tjclawson**
# Description

Add functionality to update and delete hackathons

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [X] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: app/src/main/java/com/lambdaschool/hackathon_portal/retrofit/HackathonApiInterface.kt [hunk 1]
```diff
@@ -26,4 +26,19 @@ interface HackathonApiInterface {
...
+
+    @GET("hackathons/{id}")
+    fun getHackathon(@Path("id") id: Int,
+                     @Header("Authorization") bearerToken: String): Call<Hackathon>
+
+    @PUT("hackathons/{hack_id}/u/{org_id}")
+    fun updateHackathon(@Path("hack_id") hack_id: Int,
+                        @Path("org_id") org_id: Int,
+                        @Header("Authorization") bearerToken: String,
+                        @Body hackathon: Hackathon): Call<Hackathon>
```
Comment (05nelsonm) [inline app/src/main/java/com/lambdaschool/hackathon_portal/retrofit/HackathonApiInterface.kt:39]: Fine for now, but will need to change later to use a JsonObject that is built on the fragment of only the fields needing to be updated. Maybe add a //TODO here?
```suggestion
                        // TODO: use a JsonObject of only the fields that need to be updated which
                        //  gets built on the fragment before making the method call.
                        @Body hackathon: Hackathon): Call<Hackathon>
```

File: app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/dashboard/DashboardFragment.kt [hunk 2]
```diff
@@ -89,6 +87,11 @@ class DashboardFragment : Fragment() {
         override fun onBindViewHolder(holder: ViewHolder, position: Int) {
             val data = userHackathons[position]
             holder.nameView.text = data.hackathon_name
+            holder.itemView.setOnClickListener {
+                val bundle = Bundle()
+                bundle.putInt("hackathon_id", data.hackathon_id)
+                findNavController().navigate(R.id.detailsFragment, bundle)
```
Comment (05nelsonm) [inline app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/dashboard/DashboardFragment.kt]: Injecting the NavController would be better performance wise.

adding to line 37
```
@Inject
lateinit var navController: NavController
```
```suggestion
                navController.navigate(R.id.detailsFragment, bundle)
```

File: app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/details/DetailsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                    if (hackathonId != null) {
+                        detailsViewModel.deleteHackathon(hackathonId)
+                            .observe(this, Observer {
+                                if (it != null) {
+                                    if (it) {
+                                        activity?.apply {
+                                            Toast.makeText(this,
+                                                "Successfully deleted Hackathon",
+                                                Toast.LENGTH_LONG).show()
+                                            findNavController().navigateUp()
```
Comment (05nelsonm) [inline app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/details/DetailsFragment.kt]: Injecting NavController would be better performance wise.

adding to line 34
```
@Inject
lateinit var navController: NavController
```
```suggestion
                                            navController.navigateUp()
```

**Review by 05nelsonm — CHANGES_REQUESTED** [review]
Found a couple easy improvements.

Definitely have to rack our brains on how best to address the issues with the backend's inconsistent field names for the data models.

File: app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/dashboard/DashboardFragment.kt (hunk 2 above)
Comment (tjclawson) [reply app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/dashboard/DashboardFragment.kt]: Changed in commit 770119ff532eda3f5b007d568d5ecbf612792a90

File: app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/details/DetailsFragment.kt (hunk 3 above)
Comment (tjclawson) [reply app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/details/DetailsFragment.kt]: Changed in commit 770119ff532eda3f5b007d568d5ecbf612792a90

File: app/src/main/java/com/lambdaschool/hackathon_portal/retrofit/HackathonApiInterface.kt (hunk 1 above)
Comment (tjclawson) [reply app/src/main/java/com/lambdaschool/hackathon_portal/retrofit/HackathonApiInterface.kt:39]: Added in commit 0ccde73ddeb7d91f9fefbcea8e4c1822d3ba3025

**Comment by tjclawson** [comment]
Requested changes made. User who made requests is unavailable to review. Forcing merge.

**Pull request merged**